test/functional/fixtures/regression/gh-3127/testcafe-fixtures/index-test.js

```diff
@@ -0,0 +1,47 @@
+import { stat as statCb } from 'fs';
+import { join } from 'path';
+import { homedir } from 'os';
+import { promisify } from 'util';
+import { from } from 'rxjs';
+import { delay, timeout } from 'rxjs/operators';
+import del from 'del';
+
+
+const DOWNLOADED_FILE_PATH  = join(homedir(), 'Downloads', 'package.json');
+const FILE_CHECK_INTERVAL   = 3000;
+
+const stat = promisify(statCb);
+
+async function isFile (path) {
+    try {
+        const stats = await stat(path);
+
+        return stats.isFile();
+    }
+    catch (e) {
+        return false;
+    }
+}
+
+async function waitForFile (path) {
+    while (!await isFile(path))
+        await delay(FILE_CHECK_INTERVAL);
+}
+
+async function waitForFileWithTimeout (path, ms) {
+    await from(waitForFile(path))
+        .pipe(timeout(ms))
+        .toPromise();
+}
```

Comment:
Should we use `rxjs` only for this purpose? I do not insist but I think this ` from/toPromise` code looks unobvious

Reply:
You're right, I've rewritten this part of the code.